tools/godotcpp.py

```diff
@@ -259,7 +297,10 @@ def generate(env):
     if env["use_hot_reload"]:
         env.Append(CPPDEFINES=["HOT_RELOAD_ENABLED"])
 
-    tool = Tool(env["platform"], toolpath=["tools"])
+    if env["platform"] in platforms:
+        tool = Tool(env["platform"], toolpath=["tools"])
+    else:
+        tool = Tool(env["platform"], toolpath=[get_custom_tools_path(env)])
```

Comment:
We can do this to support overriding tools for default platforms:

```suggestion
    custom_platforms = get_custom_platforms(env)
    if env["platform"] in custom_platforms:
        tool = Tool(env["platform"], toolpath=[get_custom_tools_path(env)])
    else:
        tool = Tool(env["platform"], toolpath=["tools"])
```

Reply:
Hm. Given that this is advanced functionality, I guess it probably makes sense to allow overriding the tools for default platforms? And especially since it seems like this is targeted at "users" of a GDExtension (as opposed to the "author" of a GDExtension, who could just hack up the SConstruct file), overriding could be useful. So, I guess I'd personally vote for the alternative given here.

Reply:
Applied

Reply:
Well, I ended up finding a more elegant solution for overrides, which get rid of the if/else and use the SCons `toolpath` option which already supports this feature.